Lib/test/test_typing.py

```diff
@@ -559,6 +560,20 @@ def test_no_multiple_subscripts(self):
         with self.assertRaises(TypeError):
             Literal[1][1]
 
+    def test_equal(self):
+        self.assertNotEqual(Literal[0], Literal[False])
+        self.assertNotEqual(Literal[True], Literal[1])
+        self.assertNotEqual(Literal[1], Literal[2])
+        self.assertNotEqual(Literal[1, True], Literal[1])
+        self.assertEqual(Literal[1], Literal[1])
+        self.assertEqual(Literal[1, 2], Literal[2, 1])
+        self.assertEqual(Literal[1, 2, 3], Literal[1, 2, 3, 3])
+
+    def test_flatten(self):
+        self.assertEqual(Literal[Literal[1], Literal[2], Literal[3]], Literal[1, 2, 3])
+        self.assertEqual(Literal[Literal[1, 2], 3], Literal[1, 2, 3])
+        self.assertEqual(Literal[Literal[1, 2, 3]], Literal[1, 2, 3])
```

Comment:
Again, maybe some checks verifying `__args__`?

Reply:
Added tests to cover `Literal.__args__`